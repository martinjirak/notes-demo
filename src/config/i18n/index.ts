import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import xhr from 'i18next-xhr-backend'
import { initReactI18next } from 'react-i18next'

const options: i18n.InitOptions = {
    load: 'languageOnly',
    ns: ['global'],
    fallbackLng: 'en',
    defaultNS: 'global',
    backend: {
        loadPath: 'public/locales/{{lng}}/{{ns}}.json',
    },
    detection: {
        order: ['localStorage', 'navigator'],
        caches: ['localStorage'],
        lookupLocalStorage: 'i18nextLng',
    },
    interpolation: {
        escapeValue: false,
    },
    react: {
        bindI18n: 'languageChanged',
    },
}

export default i18n.use(LanguageDetector).use(xhr).use(initReactI18next).init(options)
