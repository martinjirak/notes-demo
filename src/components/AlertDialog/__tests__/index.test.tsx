import * as React from 'react'
import { createMount } from '@material-ui/core/test-utils'
import { I18nextProvider } from 'react-i18next'
import i18n from '../../../config/tests/i18n/index'
import { AlertDialog, AlertDialogProps } from '../'

describe('AlertDialog should match Snapshot', () => {
    it('without hint', () => {
        const props: AlertDialogProps = {
            onAgree: jest.fn(),
            onClose: jest.fn(),
            text: 'Alert text',
        }

        const renderedValue = createMount()(
            <I18nextProvider i18n={i18n}>
                <AlertDialog {...props} />
            </I18nextProvider>
        )
        expect(renderedValue.html()).toMatchSnapshot()
    })
})
