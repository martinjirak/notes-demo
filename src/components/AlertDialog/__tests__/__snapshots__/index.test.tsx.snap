// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog should match Snapshot without hint 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\"><div class=\\"MuiDialogContent-root\\"><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Alert text</p></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Yes</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">No</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
